test/support/integration/teams_server.ex

```diff
@@ -0,0 +1,299 @@
+defmodule Livebook.TeamsServer do
+  @moduledoc false
+  use GenServer
+
+  defstruct [:node, :token, :user, :org, :teams_key, :port, :app_port, :url, :env]
+
+  @name __MODULE__
+  @timeout 10_000
+  @default_teams_dir "../hub"
+
+  def available?() do
+    System.get_env("TEAMS_PATH") != nil or File.exists?(@default_teams_dir)
+  end
+
+  def start(opts \\ []) do
+    GenServer.start(__MODULE__, opts, name: @name)
+  end
+
+  def url() do
+    GenServer.call(@name, :fetch_url, @timeout)
+  end
+
+  def token() do
+    GenServer.call(@name, :fetch_token, @timeout)
+  end
+
+  def user() do
+    GenServer.call(@name, :fetch_user, @timeout)
+  end
+
+  def get_node() do
+    GenServer.call(@name, :fetch_node, @timeout)
+  end
+
+  def drop_database() do
+    app_port = GenServer.call(@name, :fetch_port)
+    state_env = GenServer.call(@name, :fetch_env)
+
+    app_port |> env(state_env) |> mix(["ecto.drop", "--quiet"])
+  end
+
+  # GenServer Callbacks
+
+  @impl true
+  def init(opts) do
+    state = struct!(__MODULE__, opts)
+    {:ok, %{state | node: app_node()}, {:continue, :start_app}}
+  end
+
+  @impl true
+  def handle_continue(:start_app, state) do
+    ensure_app_dir!()
+    prepare_database(state)
+
+    {:noreply, %{state | port: start_app(state)}}
+  end
+
+  @impl true
+  def handle_call(:fetch_token, _from, state) do
+    state = if state.token, do: state, else: ensure_session_token(state)
+
+    {:reply, state.token, state}
+  end
+
+  @impl true
+  def handle_call(:fetch_user, _from, state) do
+    state = if state.user, do: state, else: ensure_user(state)
+
+    {:reply, state.user, state}
+  end
+
+  @impl true
+  def handle_call(:fetch_url, _from, state) do
+    state = if state.app_port, do: state, else: %{state | app_port: app_port()}
+    url = state.url || fetch_url(state)
+
+    {:reply, url, %{state | url: url}}
+  end
+
+  def handle_call(:fetch_node, _from, state) do
+    {:reply, state.node, state}
+  end
+
+  def handle_call(:fetch_port, _from, state) do
+    app_port = state.app_port || app_port()
+
+    {:reply, app_port, %{state | app_port: app_port}}
+  end
+
+  def handle_call(:fetch_env, _from, state) do
+    {:reply, state.env, state}
+  end
+
+  # Port Callbacks
+
+  @impl true
+  def handle_info({_port, {:data, message}}, state) do
+    info(message)
+    {:noreply, state}
+  end
+
+  def handle_info({_port, {:exit_status, status}}, _state) do
+    error("enterprise quit with status #{status}")
+    System.halt(status)
+  end
+
+  # Private
+
+  defp call_erpc_function(node, function, args \\ []) do
+    :erpc.call(node, Hub.Integration, function, args)
+  end
+
+  defp ensure_session_token(state) do
+    state =
+      state
+      |> ensure_user()
+      |> ensure_org()
+      |> ensure_teams_key()
```

Comment:
ditto, why don't we do this in init/1 once?

Reply:
In this case, no, because we need to wait until the server starts (migrations up and phoenix start), so we can start executing the functions through `erpc.call/4`